presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGenerator.java

```diff
@@ -392,6 +395,68 @@ else if (fractionInput instanceof VariableReferenceExpression) {
             return format("PERCENTILEEST%d(%s)", percentile, inputSelections.get(getVariableReference(inputs.get(0))));
         }
 
+        private String handleApproxDistinct(CallExpression aggregation, Map<VariableReferenceExpression, Selection> inputSelections)
+        {
+            List<RowExpression> inputs = aggregation.getArguments();
+            if (inputs.isEmpty() || inputs.size() > 2) {
+                throw new PinotException(PINOT_UNSUPPORTED_EXPRESSION, Optional.empty(), "Cannot handle approx_distinct function " + aggregation);
+            }
+            Selection selection = inputSelections.get(getVariableReference(inputs.get(0)));
+            if (inputs.size() == 1) {
+                return format("DISTINCTCOUNTHLL(%s)", selection);
+            }
+            RowExpression standardErrorInput = inputs.get(1);
+            String standardErrorString;
+            if (standardErrorInput instanceof ConstantExpression) {
+                standardErrorString = getLiteralAsString((ConstantExpression) standardErrorInput);
+            }
+            else if (standardErrorInput instanceof VariableReferenceExpression) {
+                Selection fraction = inputSelections.get(standardErrorInput);
+                if (fraction.getOrigin() != LITERAL) {
+                    throw new PinotException(PINOT_UNSUPPORTED_EXPRESSION, Optional.empty(),
```

Comment:
one param per line; leave the first line empty

Reply:
done.